src/mobile/src/main/java/tupperdate/android/home/Home.kt

```diff
@@ -1,98 +1,146 @@
-        modifier.padding(10.dp)
-    ) {
-        Text(text = "Hi there ${user.displayName}")
-        Button(onClick = onButtonClick) {
-            Text("View Branding")
+    Row(Modifier.fillMaxSize()) {
+        Column(
+            Modifier.fillMaxWidth().align(Alignment.Bottom)
+                .padding(bottom = veryLittleButtonSize.dp)
+        ) {
+            WithConstraints { -> //why no parameter here ??
+
+                //determine height and width of card according to constraints
+                //the card measures 90 % of screen width and 75 % of screen height
+                val boxHeight =
+                    with(DensityAmbient.current) { constraints.maxHeight.toDp() * 0.75f }
+                val boxWidth = with(DensityAmbient.current) { constraints.maxWidth.toDp() * 0.9f }
+
+                //the align(..) doesn't work anymore (because of reasons), so we need to
+                //set manually a default pos
+                val defaultPos = ((with(DensityAmbient.current) { constraints.maxWidth.toDp() / 2 })
+                        - boxWidth / 2).value.toInt()
+                val (pos, setPos) = remember { mutableStateOf((defaultPos)) }
+
+                //the defaultPos needs a delta measured
+                val (delta, setDelta) = remember { mutableStateOf(0) }
+                val swipeMargin = (boxWidth).value.toInt()
+
+                Box(modifier = Modifier.align(Alignment.CenterHorizontally)
+                    .width((boxWidth)).height((boxHeight))
+                    .padding(bottom = (veryLittleButtonSize * 0.8f).dp)
+                    .draggable(Orientation.Horizontal,
+                        onDragStarted = {}, onDrag =
+                        { deltaTemp ->
+                            run {
+                                setPos(pos + (deltaTemp * 0.75).toInt())
+                                setDelta(delta+deltaTemp.toInt())
+                            }
+                        }, onDragStopped = {
+                            if (delta.absoluteValue < swipeMargin) {
+                                setPos(defaultPos)
+                                setDelta(0)
+                            }
+                        })
+                )
+                {
+                    if (delta.absoluteValue > swipeMargin) {
+                        if (pos > defaultPos) {
+                            onLike()
+                            Log.d("HOMe", "like")
+                        } else if (pos < defaultPos) {
+                            onDislike()
+                            Log.d("HOMe", "dislike")
+                        }
+                        setPos(defaultPos)
+                        setDelta(0)
+                        //the like() call will change the presentRecipe
+                        recipeCard(pos, presentRecipe, boxWidth)
+                    } else {
+                        recipeCard(pos, presentRecipe, boxWidth)
+                        Log.d("HOMe", "normal")
```

Comment:
Do not call callbacks from within composable functions. Use callbacks from the DragListener, for instance, but do NEVER rely on the fact that composable bodies are executed only once (because they are not).